acvm-repo/acvm/tests/solver.rs

```diff
@@ -319,6 +321,7 @@ fn double_inversion_brillig_oracle() {
         witness_assignments,
         &unconstrained_functions,
         &[],
+        false,
```

Comment:
If it's mostly `false`, maybe it would make sense to add a `fn with_profiler(self, enable: bool) -> Self` method to selectively turn it on. It might even return a special wrapper that allows you to call a version of `finalize` that returns an `Option<Samples>`, but the base VM would not collect samples and then no panics are necessary.

Reply:
I added this method.